### PR TITLE
[SYSADMIN ACTION][Feature:Installation] static maintenance webpage

### PR DESCRIPTION
**Pull request by zjxiaohan**
To take advantage of this feature, it is necessary to modify the apache configuration file.
Detailed instructions will be posted here:
https://submitty.org/sysadmin/version_notes/v19.08.03
For upcoming release:
https://github.com/Submitty/Submitty/releases/tag/v19.08.03

While running INSTALL_SUBMITTY.sh to update the submitty source code (~1-2 minutes), there will be a brief interruption in serving all webpages on the site.  To avoid confusing errors, we place a simple static maintenance webpage instructing the user to try again in a couple minutes.

Closes https://github.com/Submitty/Submitty/issues/4350

Implemented as suggested by @MasterOdin

### CODE REVIEW

File: .setup/INSTALL_SUBMITTY_HELPER_SITE.sh [hunk 1]
```diff
@@ -149,8 +153,4 @@ chmod 550 ${SUBMITTY_INSTALL_DIR}/site/cgi-bin/git-http-backend
 # cache needs to be writable
 find ${SUBMITTY_INSTALL_DIR}/site/cache -type d -exec chmod u+w {} \;
 
-# return the course index page (only necessary when 'clean' option is used)
```
Comment (zjxiaohan) [inline .setup/INSTALL_SUBMITTY_HELPER_SITE.sh]: This is removed as a continuing part of https://github.com/Submitty/Submitty/pull/4337

**Comment by MasterOdin** [comment]
@zjxiaohan I made some changes to the branch as there was a couple things I didn't think about:
1. on initial installation, `site/public` does not exist so have to make it
1. have to insert a rewrite rule for when index.html exists in apache so that the new URLs (like `/home`) would serve `index.html` instead of `index.php`.

**Comment by zjxiaohan** [comment]
Rewrite rule is something I didn't consider and now I know that my new URLs were not interrupted during local tests because they were redirected to `index.php`. Thanks for editing this!

**Comment by MasterOdin** [comment]
These two lines however cause a 403 Forbidden error and need to be updated:
https://github.com/Submitty/Submitty/blob/9634054480db5f12e420ab4af23bbde9b619dfd3/.setup/INSTALL_SUBMITTY_HELPER_SITE.sh#L120-L124

I think what we might want to do here is change this to be a find over all files/directories to change the permissions, but to skip `site/public/index.html` (as that needs to retain the other read bit). Additionally, we might want to change it such that we iterate first over all directories giving them the appropriate permissions (551), and then iterate over files giving them the right permissions (440).

**Comment by MasterOdin** [comment]
Checked that the permissions did not change on any files by applying this patch by doing:
```
cd /usr/local/submitty/site
find . -exec stat --format='%n %A %U %G' {} \; | sort
```

and using `diff` on the output from before running this and after.

**Pull request merged**